pyrobosim/pyrobosim/utils/search_graph.py

```diff
@@ -1,6 +1,68 @@
 """ Graph search utilities. """
 
+import warnings
+
 import numpy as np
+from astar import AStar
+
+
+class SearchGraphPlanner(AStar):
```

Comment:
Can this all be part of the `SearchGraph` class itself, if you inherit it from `AStar` directly?

Reply:
It could, but I went this way since I like the separation of planner and graph.

Reply:
If that's the case, then let's make the separation cleaner.

Right now, when you instantiate a `SearchGraph`, you always get a `SearchGraphPlanner` object inside of it as the `path_finder` attribute.

Maybe instead have `SearchGraphPlanner` accept a `SearchGraph` in its constructor, so 
* The planner contains a graph
* but the graph doesn't necessarily contain a planner

Reply:
> Maybe instead have SearchGraphPlanner accept a SearchGraph

Had thought about this, but did not feel intuitive since it is only used by the graph internally through the `find_path` method.
I have now updated it so that there is a `use_planner` argument for the graph which is `False` by default.
So when the caller only needs a graph, it gets a graph and when called with `use_planner=True` , it will will graph with planner embedded.